src/services/jobs.js

```diff
@@ -16,14 +21,20 @@ const getStaffPageForUser = async () => {
 /**
  * Get active jobs for current user
  * @param {Boolean} canUseStaff Whether user can use staff timesheets
- * @param {String} details The user's details
+ * @param {Date} shiftStart The start of the shift to get jobs for
+ * @param {Date} shiftEnd The end of the shift to get jobs for
  * @return {Promise.<String>} User's active jobs
  */
-const getActiveJobsForUser = (canUseStaff, details) => {
+const getJobs = (canUseStaff, shiftStart, shiftEnd) => {
+  const urlParams = `?shiftStart=${dateFormat.format(shiftStart)}&shiftEnd=${dateFormat.format(
```

Comment:
With the apartment application page, we handle multi-parameter route calls this way:  In the API, we create a view model that contains a class variable for each parameter and replace all parameters in the controller with one of the created view model. Then, the frontend just needs to pass a JSON string with a similar format to the API view model.
I do not know if this is common across the whole API, but it may be better and more readable than this URL syntax.

Reply:
I agree that this interpolated string is not the clearest thing in the world, but I think it is readable and the query parameter names provide enough context to make the rest fairly obvious. The main reason we can't use a View Model and send a complex type in the reuqest is that this is a GET method, which means it should never have a body (or more specifically, by HTTP specification, any body in a GET request should be ignored). See the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) for reference. The right way to send request parameters in a GET method is using query parameters, which is exactly what is being done here. This is the standard and accepted way of interacting with a REST API.

The old, deprecated route for getting a user's jobs (`HTTP POST /api/jobs/getJobs`) included a `details` object in the parameter. However, as you can see this was perhaps even more confusing and broke REST conventions by using a POST method to request information without creating any information, and also it had a DIFFERENT HTTP method specified in the URL route, which was both unconventional and confusingly contradictory. I'm inclined to believe the revised approach using query parameters is clearer overall.

Reply:
I see! Definitely keep it that way then.